test/unit/clients/connector_client_google_authentication_tests.js

```diff
@@ -0,0 +1,137 @@
+'use strict'
+
+// Core dependencies
+const path = require('path')
+
+// NPM dependencies
+const Pact = require('pact')
+const chai = require('chai')
+const chaiAsPromised = require('chai-as-promised')
+
+// Constants
+const TEST_CHARGE_ID = 'testChargeId'
+const GOOGLE_AUTH_PATH = `/v1/frontend/charges/${TEST_CHARGE_ID}/wallets/google`
+const PORT = Math.floor(Math.random() * 48127) + 1024
+const BASEURL = `http://localhost:${PORT}`
+
+// Custom dependencies
+const connectorClient = require('../../../app/services/clients/connector_client')
+const fixtures = require('../../fixtures/wallet_payment_fixtures.js')
+const { PactInteractionBuilder } = require('../../fixtures/pact_interaction_builder')
+
+// Global setup
+const expect = chai.expect
+chai.use(chaiAsPromised)
+
+describe('connectors client - google authentication API', function () {
+  const provider = Pact({
+    consumer: 'frontend',
+    provider: 'connector',
+    port: PORT,
+    log: path.resolve(process.cwd(), 'logs', 'mockserver-integration.log'),
+    dir: path.resolve(process.cwd(), 'pacts'),
+    spec: 2,
+    pactfileWriteMode: 'merge'
+  })
+
+  before(() => provider.setup())
+  after((done) => provider.finalize().then(done()))
+
+  describe('Authenticate google payment', function () {
+    describe('authorisation success', function () {
+      const successfulGoogleAuthRequest = fixtures.googleAuthRequestDetails()
+      const authorisationSuccessResponse = fixtures.googleAuthSuccessResponse()
+
+      before((done) => {
+        provider.addInteraction(
+          new PactInteractionBuilder(GOOGLE_AUTH_PATH)
+            .withRequestBody(successfulGoogleAuthRequest.getPactified())
+            .withMethod('POST')
+            .withState('a charge exists with id testChargeId and is in state ENTERING_CARD_DETAILS.')
```

Comment:
sorry to be a pain but after chatting with @danworth i think this state should indicate the charge is using the sandbox payment provider as you're using the fixtures applicable for sandbox to drive the response behaviour from connector.

Reply:
Yes, this is a good suggestion since we're reliant on the account being set-up for `sandbox` 👍 I'll update.